### PR TITLE
version upgrade

### PR DESCRIPTION
**Pull request by kumarabd**
Signed-off-by: kumarabd <abishekkumar92@gmail.com>

**Description**

- Broker package created
- Config package moved from adapter library
- Meshery kubeclient upgraded with dynamic client
- removed smi package
- DetectKubeConfig() Updated with input support

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: broker/broker.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	Add    EventType = "ADDED"
+	Update EventType = "MODIFIED"
+	Delete EventType = "DELETED"
+	Error  EventType = "ERROR"
+)
+
+type ObjectType string
+type EventType string
+
+type Message struct {
```
Comment (leecalcote) [inline broker/broker.go:16]: A note to be working toward a common message struct - https://docs.google.com/document/d/1_zpEWBcC6ngOLen_E_nc-octiUdBbpmdwwIX6qzo70M/edit#heading=h.3gmuxn6p14m9

File: config/config.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 Layer5, Inc.
```
Comment (leecalcote) [inline config/config.go:1]:
```suggestion
// Copyright 2021 Layer5, Inc.
```

File: config/error.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 Layer5, Inc.
```
Comment (leecalcote) [inline config/error.go:1]:
```suggestion
// Copyright 2021 Layer5, Inc.
```

File: config/provider/inmem.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Layer5, Inc.
```
Comment (leecalcote) [inline config/provider/inmem.go:1]:
```suggestion
// Copyright 2021 Layer5, Inc.
```

File: config/provider/provider.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 Layer5, Inc.
```
Comment (leecalcote) [inline config/provider/provider.go:1]:
```suggestion
// Copyright 2021 Layer5, Inc.
```

File: config/provider/viper.go [hunk 6]
```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Layer5, Inc.
```
Comment (leecalcote) [inline config/provider/viper.go:1]:
```suggestion
// Copyright 2021 Layer5, Inc.
```

File: utils/kubernetes/error.go [hunk 7]
```diff
@@ -2,6 +2,13 @@ package kubernetes
 
 import "github.com/layer5io/meshkit/errors"
 
+const (
+	ErrNewKubeClientCode = "test_code"
```
Comment (leecalcote) [inline utils/kubernetes/error.go:6]: MESHSYNC1000 maybe?

File: utils/kubernetes/error.go [hunk 8]
```diff
@@ -15,3 +22,23 @@ func ErrServiceDiscovery(err error) error {
 func ErrApplyHelmChart(err error) error {
 	return errors.NewDefault(errors.ErrApplyHelmChart, "Error applying helm chart: "+err.Error())
 }
+
+// ErrApplyHelmChart is the error which occurs in the process of applying helm chart
+func ErrNewKubeClient(err error) error {
```
Comment (leecalcote) [inline utils/kubernetes/error.go:27]: @kumarabd do you have an example of one of these also providing probable cause and suggested remediation?

**Review by leecalcote — APPROVED** [review]
A couple of items to note, but nothing blocking. LGTM

File: utils/kubernetes/error.go (hunk 7 above)
Comment (kumarabd) [reply utils/kubernetes/error.go:6]: Right now this is not being used anywhere. May be after we decide on the error code strategy? @leecalcote

File: utils/kubernetes/error.go (hunk 8 above)
Comment (kumarabd) [reply utils/kubernetes/error.go:27]: Same here, we're using a temp method which doesn't have cause and other properties. We can switch to that when we are final with the strategy.

**Pull request merged**